site/src/component/SearchHitContainer/SearchHitContainer.tsx

```diff
@@ -22,15 +22,33 @@ const SearchResults = ({
   CourseHitItem,
   ProfessorHitItem,
 }: Required<SearchHitContainerProps> & { results: CourseGQLData[] | ProfessorGQLData[] }) => {
+  const roadmap = useAppSelector((state) => state.roadmap);
+  const allExistingCourses = roadmap?.yearPlans.flatMap((yearPlan) =>
+    yearPlan.quarters.flatMap((quarter) => quarter.courses),
+  );
```

Comment:
Needs a fix for the multiplanner.

```suggestion
  const allExistingCourses = roadmap?.plans[roadmap.currentPlanIndex].content.yearPlans.flatMap((yearPlan) =>
    yearPlan.quarters.flatMap((quarter) => quarter.courses),
  );
```